src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,330 @@
+  g3 = tau_22 * mu
+  g4 = ( v1 * tau_12 + v2 * tau_22 + q2 ) * mu
+
+  return (SVector(f1, f2, f3, f4) , SVector(g1, g2, g3, g4))
+end
+
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::CompressibleNavierStokesEquations2D)
+  rho, rho_v1, rho_v2, _ = u
+
+  v1 = rho_v1 / rho
+  v2 = rho_v2 / rho
+  T  = temperature(u, equations)
+
+  return SVector(rho, v1, v2, T)
+end
+
+# TODO: make this consistent with cons2prim above and cons2prim for Euler!
+@inline prim2cons(u, equations::CompressibleNavierStokesEquations2D) =
+    prim2cons(u, equations.equations_hyperbolic)
+
+
+# # Convert conservative variables to entropy
+# @inline function cons2entropy(u, equations::CompressibleNavierStokesEquations2D)
+#   rho, rho_v1, rho_v2, rho_e = u
+
+#   v1 = rho_v1 / rho
+#   v2 = rho_v2 / rho
+#   v_square = v1^2 + v2^2
+#   p = (equations.gamma - 1) * (rho_e - 0.5 * rho * v_square)
+
+#   rho_p = rho / p
+
+#   w2 = rho_p * v1
+#   w3 = rho_p * v2
+#   w4 = -rho_p
+
+#   return SVector(w2, w3, w4)
+# end
+
+
+# @inline function convert_gradient_variables(u, grad_entropy_vars, equations::CompressibleNavierStokesEquations2D)
+# # Takes the solution values `u` and gradient of the variables (w_2, w_3, w_4) and
+# # reverse engineers the gradients to be terms of the primitive variables (v1, v2, T).
+# # Helpful because then the diffusive fluxes have the same form as on paper.
+#   rho, rho_v1, rho_v2, _ = u
+
+#   v1 = rho_v1 / rho
+#   v2 = rho_v2 / rho
+#   T  = temperature(u, equations)
+
+#   return SVector(equations.R * T * (grad_entropy_vars[1] + v1 * grad_entropy_vars[3]), # grad(u) = R*T*(grad(w_2)+v1*grad(w_4))
+#                  equations.R * T * (grad_entropy_vars[2] + v2 * grad_entropy_vars[3]), # grad(v) = R*T*(grad(w_3)+v2*grad(w_4))
+#                  equations.R * T * T * grad_entropy_vars[3]                            # grad(T) = R*T^2*grad(w_4))
+#                 )
+# end
+
+
+@inline function temperature(u, equations::CompressibleNavierStokesEquations2D)
+  rho, rho_v1, rho_v2, rho_e = u
+
+  p = (equations.gamma - 1) * (rho_e - 0.5 * (rho_v1^2 + rho_v2^2) / rho)
+  T = p / (equations.R * rho)
+  return T
+end
+
+"""
+    struct BoundaryConditionViscousWall{V, H}
+
+Creates a wall-type boundary conditions for the compressible Navier-Stokes equations.
+The fields `boundary_condition_velocity` and `boundary_condition_heat_flux` are intended
+to be boundary condition types such as the `NoSlip` velocity boundary condition and the
+`Adiabatic` or `Isothermal` heat boundary condition.
+"""
+struct BoundaryConditionViscousWall{V, H}
+  boundary_condition_velocity::V
+  boundary_condition_heat_flux::H
+end
+
+"""
+    struct NoSlip{F}
+
+Creates a no-slip boundary condition with field `boundary_value_function`, which
+should be a function with signature `boundary_value_function(x, t, equations)`
+and should return a `SVector{NDIMS}` whose entries are the velocity vector at a
+point `x` and time `t`.
```

Comment:
```suggestion
Use to create a no-slip boundary condition with `BoundaryConditionViscousWall`. The field `boundary_value_function`
should be a function with signature `boundary_value_function(x, t, equations)`
and should return a `SVector{NDIMS}` whose entries are the velocity vector at a
point `x` and time `t`.
```
I suggest a similar change to the other `Adiabatic` and `Isothermal`, as it now reads as if I just create, e.g., a `NoSlip` struct and then this would be the full boundary condition.